### PR TITLE
Fix filter in Adapt.findRelativeModel

### LINKED ISSUE #3031: Filter is overzealous in Adapt.findRelativeModel
### Subject of the issue/enhancement/features
`options.filter` in `Adapt.findRelativeModel` erroneously filters out the current model producing inconsistencies:
https://github.com/adaptlearning/adapt_framework/blob/5a3de6952f016e70aacd986a9e6c71e0ebed462b/src/core/js/models/adaptModel.js#L499-L502

### Your environment
Framework 5.10

### Steps to reproduce
In the core-bundled course, run the following in the console to make co-10 optional and attempt to retrieve its previous (mandatory) page ID:
```js
require('core/js/adapt').findById('co-10')
  .set('_isOptional', true)
  .findRelativeModel('contentobject -1', {
    filter: model => !model.get('_isOptional')
  }).get('_id');
```

### Expected behaviour
`"co-05"` should be returned from the above code.

### Actual behaviour
The following is returned because co-10 filters itself out:
```console
adaptModel.js:527 Uncaught TypeError: Cannot read property 'isTypeGroup' of undefined
    at PageModel.findRelativeModel (adaptModel.js:527)
    at <anonymous>:3:4
```

### PR DESCRIPTION
**Pull request by tomgreenfield**
Resolves #3031.

### CODE REVIEW

File: src/core/js/models/adaptModel.js [hunk 1]
```diff
@@ -524,17 +523,18 @@ export default class AdaptModel extends LockingModel {
...
-          // there is no descendant which matches this relativeString
-          // probably looking for the descendant 0 in a parent
-          break;
-        }
-        if (movementCount === moveBy) {
-          return Adapt.findById(descendant.get('_id'));
-        }
-        movementCount++;
+      if (!descendant.isTypeGroup(relativeDescriptor.type)) continue;
+      const isSelf = i === modelIndex;
```
Comment (oliverfoster) [inline src/core/js/models/adaptModel.js:527]: Can I get some brackets up in here?
```suggestion
      const isSelf = (i === modelIndex);
```

File: src/core/js/models/adaptModel.js [hunk 2]
```diff
@@ -510,6 +505,10 @@ export default class AdaptModel extends LockingModel {
       return false;
     });
 
+    const shouldSkip = model => {
```
Comment (moloko) [inline src/core/js/models/adaptModel.js:508]: comment with link to issue might be useful here i.e.
// https://github.com/adaptlearning/adapt_framework/issues/3031

File: src/core/js/models/adaptModel.js [hunk 3]
```diff
@@ -510,6 +505,10 @@ export default class AdaptModel extends LockingModel {
...
+      return typeof options.filter === 'function' && !options.filter(model);
```
Comment (oliverfoster) [inline src/core/js/models/adaptModel.js:509]: I think I preferred your original version where `typeof options.filter === 'function'` wasn't executed at each iteration.

**Review by oliverfoster — CHANGES_REQUESTED** [review]

File: src/core/js/models/adaptModel.js [hunk 4]
```diff
@@ -504,10 +504,7 @@ export default class AdaptModel extends LockingModel {
       }
       return false;
     });
-
-    const shouldSkip = model => {
-      return typeof options.filter === 'function' && !options.filter(model);
-    };
+    const isFilterFunction = typeof options.filter === 'function';
```
Comment (oliverfoster) [inline src/core/js/models/adaptModel.js:507]:
```suggestion
    const isFilterFunction = (typeof options.filter === 'function');
```

**Review by moloko — APPROVED** [review]
code 👓

**Review by oliverfoster — APPROVED** [review]

**Review by eleanor-heath — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Pull request merged**